### PR TITLE
feat(components/forms): add `hintText` to file attachment

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2769089](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2769089)

BEGIN_COMMIT_OVERRIDE
feat(components/forms): add hintText to file attachment
END_COMMIT_OVERRIDE

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2241/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2241/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2241/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2241/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2241/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2241/playground/)

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts [hunk 1]
```diff
@@ -50,6 +71,21 @@ export class FileAttachmentComponent {
...
 
+  protected onFileChange(result: SkyFileAttachmentChange): void {
+    const file = result.file;
+
+    if (file && file.errorType) {
+      this.#reactiveFile?.setValue(undefined);
+      // this.reactiveUploadError = this.#getErrorMessage(
+      //   file.errorType,
+      //   file.errorParam,
+      // );
```
Comment (Blackbaud-SteveBrush) [inline apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts]: Commented-out code?

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.html [hunk 2]
```diff
@@ -16,3 +16,15 @@
...
+<form [formGroup]="formGroup">
+  <sky-file-attachment
+    required="true"
+    labelText="Single file attachment"
+    hintText="Sample hint text"
+    [acceptedTypes]="acceptedTypes"
+    [maxFileSize]="maxFileSize"
+    [validateFn]="validateFile"
+    (fileChange)="onFileChange($event)"
+  ></sky-file-attachment>
```
Comment (Blackbaud-SteveBrush) [inline apps/playground/src/app/components/forms/file-attachment/file-attachment.component.html]: Can be self-closing. Also, can you check the alphabetical order of the attributes?

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.module.ts [hunk 3]
```diff
@@ -16,6 +16,8 @@ import { FileAttachmentComponent } from './file-attachment.component';
 @NgModule({
   imports: [
     CommonModule,
+    FormsModule,
+    ReactiveFormsModule,
```
Comment (Blackbaud-SteveBrush) [inline apps/playground/src/app/components/forms/file-attachment/file-attachment.module.ts]: Alphabetical order.

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts [hunk 4]
```diff
@@ -20,11 +31,21 @@ export class FileAttachmentComponent {
   public rejectedFiles: SkyFileItem[];
   protected required = true;
 
+  protected attachment: FormControl;
+  protected formGroup: FormGroup;
```
Comment (Blackbaud-SteveBrush) [inline apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts:34]: ☝️ Alphabetical order.

File: libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts [hunk 5]
```diff
@@ -119,6 +119,13 @@ export class SkyFileAttachmentComponent
   @Input({ transform: booleanAttribute })
   public labelHidden = false;
 
+  /**
+   * [Persistent inline help text](https://developer.blackbaud.com/skyux/design/guidelines/user-assistance#inline-help) that provides
+   * additional context to the user.
+   */
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts:126]: I believe we want the `@preview` tag on this?

Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts:126]: added

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts (hunk 4 above)
Comment (Blackbaud-CoreyArcher) [reply apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts:34]: reordered

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.module.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply apps/playground/src/app/components/forms/file-attachment/file-attachment.module.ts]: fixed

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.html (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply apps/playground/src/app/components/forms/file-attachment/file-attachment.component.html]: updated

File: apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply apps/playground/src/app/components/forms/file-attachment/file-attachment.component.ts]: removed

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**